### PR TITLE
Khweb 31

### PR DESCRIPTION
**Pull request by SergeySMS**

### CODE REVIEW

File: src/app/components/profile-edit/profile-edit.component.ts [hunk 1]
```diff
@@ -1,29 +1,71 @@
...
+            .select('editProfile')
+            .pipe(takeUntil(this.subscription$))
+            .subscribe(inf => {
+                this.profileForm.value.userName = inf.name;
+                this.profileForm.value.userSurname = inf.surname;
+                this.url = this.defaultImage;
+            });
+    }
+    ngAfterViewChecked() {
+        this.cdRef.detectChanges();
```
Comment (didevgen) [inline src/app/components/profile-edit/profile-edit.component.ts]: bad idea as for me

File: src/app/components/profile-edit/profile-edit.component.ts [hunk 2]
```diff
@@ -1,29 +1,71 @@
...
+    }
+    ngOnDestroy() {
+        this.subscription$.next();
```
Comment (didevgen) [inline src/app/components/profile-edit/profile-edit.component.ts]: ??? there must be unsubscriptions

File: src/app/components/profile-edit/profile-edit.service.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    userID: string;
+
+    constructor(
+        private storage: AngularFireStorage,
+        private db: AngularFirestore,
+    ) {}
+
+    saveData(payload: EditProfile): Observable<EditProfile> {
+        this.data = payload;
+        this.userID = '200 OK'; // should be user id from collection
```
Comment (didevgen) [inline src/app/components/profile-edit/profile-edit.service.ts]: ???

File: src/app/components/profile-edit/profile-edit.service.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+        const docR = this.db.doc('Users/' + this.userID);
```
Comment (didevgen) [inline src/app/components/profile-edit/profile-edit.service.ts]: what is docR? where is docB, docZ?

File: src/app/components/profile-edit/store/profile-edit.effects.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+import { of } from 'rxjs';
+import * as EdProfActions from './profile-edit.actions';
+import { ProfileEditService } from '../profile-edit.service';
+
+@Injectable()
+export class ProfileEditEffect {
+    loadEditProfile$ = createEffect(() =>
+        this.actions$.pipe(
+            ofType(EdProfActions.ProfileEditTypes.ProfileEdit_Set),
+            switchMap((act: EdProfActions.ProfileEditSetSuccess) =>
```
Comment (didevgen) [inline src/app/components/profile-edit/store/profile-edit.effects.ts]: EdProfActions?

File: src/app/components/profile-edit/store/profile-edit.effects.ts [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+                            }),
+                    ),
+                    catchError(err => {
+                        return of(new EdProfActions.ProfileEditSetError());
+                    }),
+                ),
+            ),
+        ),
+    );
+    constructor(
```
Comment (didevgen) [inline src/app/components/profile-edit/store/profile-edit.effects.ts]: constructor is always first member

File: src/app/components/profile-edit/store/profile-edit.reducer.ts [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    surname: null,
+    avatar: null,
+};
+
+export function profileEditReducer(
+    state: ProfileEditState = initialState,
+    action: ProfEdAct.ProfileEditActions,
+): ProfileEditState {
+    switch (action.type) {
+        case ProfEdAct.ProfileEditTypes.ProfileEdit_Set:
```
Comment (didevgen) [inline src/app/components/profile-edit/store/profile-edit.reducer.ts]: 🤦‍♂

File: src/app/components/profile-edit/store/profile-edit.reducer.ts [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+        case ProfEdAct.ProfileEditTypes.ProfileEdit_SetSuccess:
+            return {
+                ...state,
+                name: action.payload.name,
+                surname: action.payload.surname,
+                avatar: action.payload.avatar,
+            };
+        case ProfEdAct.ProfileEditTypes.ProfileEdit_SetError:
+            return {
+                ...state,
```
Comment (didevgen) [inline src/app/components/profile-edit/store/profile-edit.reducer.ts]: 🤦‍♂

File: src/app/components/profile-edit/profile-edit.component.ts (hunk 1 above)
Comment (SergeySMS) [reply src/app/components/profile-edit/profile-edit.component.ts]: was deleted

File: src/app/components/profile-edit/profile-edit.service.ts (hunk 4 above)
Comment (SergeySMS) [reply src/app/components/profile-edit/profile-edit.service.ts]: resoleved

File: src/app/components/profile-edit/store/profile-edit.effects.ts (hunk 5 above)
Comment (SergeySMS) [reply src/app/components/profile-edit/store/profile-edit.effects.ts]: resoleved

File: src/app/components/profile-edit/store/profile-edit.effects.ts (hunk 6 above)
Comment (SergeySMS) [reply src/app/components/profile-edit/store/profile-edit.effects.ts]: resoleved

File: src/app/components/profile-edit/store/profile-edit.reducer.ts (hunk 7 above)
Comment (SergeySMS) [reply src/app/components/profile-edit/store/profile-edit.reducer.ts]: wrote full name

**Comment by SergeySMS** [comment]
needs to add image and save data after refresh for google account

**Pull request merged**